src/flb_input_chunk.c

```diff
@@ -190,11 +192,37 @@ struct flb_input_chunk *flb_input_chunk_create(struct flb_input_instance *in,
         cio_chunk_down(chunk);
     }
 
+    /* Calculate the routes_mask for the input chunk */
+    chunk_routes_mask = flb_router_get_routes_mask_by_tag(tag, tag_len, in);
+    if (chunk_routes_mask == -1) {
+        flb_error("[input chunk] fail to calculate the routes_mask for input chunk");
+        cio_chunk_close(chunk, CIO_TRUE);
+        return NULL;
+    }
+    ic->routes_mask = chunk_routes_mask;
+
     return ic;
 }
 
 int flb_input_chunk_destroy(struct flb_input_chunk *ic, int del)
 {
+    ssize_t bytes;
+    uint64_t routes_mask;
+    uint64_t o_mask_id;
+    struct mk_list *head;
+    struct mk_list *tmp;
+    struct flb_output_instance *o_ins;
+
+    mk_list_foreach_safe(head, tmp, &ic->in->config->outputs) {
+        o_ins = mk_list_entry(head, struct flb_output_instance, _head);
+        o_mask_id = o_ins->mask_id;
+        routes_mask = ic->routes_mask;
```

Comment:
variable don't needed, just use ic->routes_mask in the next conditional, it never changes.

Reply:
updated